clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -774,8 +774,12 @@ public KafkaConsumer(Map<String, Object> configs,
             );
 
             // no coordinator will be constructed for the default (null) group id
-            this.coordinator = !groupId.isPresent() ? null :
-                new ConsumerCoordinator(groupRebalanceConfig,
+            if (!groupId.isPresent()) {
+                config.ignore(ConsumerConfig.AUTO_COMMIT_INTERVAL_MS_CONFIG);
+                config.ignore(ConsumerConfig.THROW_ON_FETCH_STABLE_OFFSET_UNSUPPORTED);
```

Comment:
Why the `THROW_ON_FETCH_STABLE_OFFSET_UNSUPPORTED` config is related to groupID?

Reply:
Because the value of `groupId` will affect whether to construct the` consumerCoordinator`,
Constructing `consumerCoordinator` will retrieve `THROW_ON_FETCH_STABLE_OFFSET_UNSUPPORTED` config.